### ISSUE TITLE
podman-2.0.2: --pids-limit value out of range

### ISSUE THREAD

**Issue by srcshelton** [containers/podman#6908]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I added `--pids-limit 1024` to my default launch scripts for running `podman` (after nearly accidentally fork-bombing my host!) at about `podman-1.9.3`, and this has been accepted (... and, I assume, applied?) ever since...

However, with `podman-2.0.2` suddenly I'm getting:

```
Error: strconv.ParseInt: parsing "1024": value out of range
```
... as `podman`'s sole output.

I've tried various different values, and it turns out that values from `0` to `511` are accepted, but values `512` and above result in the above error.  I've confirmed that this is the `--pids-limit` parameter causing this issue.

A value of 1024 seems sane for a root-run container, and has always been accepted up to and including `podman-2.0.1` - has this always been silently ignored, or is there a problem introduced with `podman-2.0.2`?

(Is there some `sysctl` tunable or otherwise which limits the default number of available PIDs which wasn't previously being checked?)

**Output of `podman version`:**

```
Version:      2.0.2
API Version:  1
Go Version:   go1.14.2
Git Commit:   201c9505b88f451ca877d29a73ed0f1836bb96c7
Built:        Wed Jul  8 13:06:00 2020
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.15.0
  cgroupVersion: v1
  conmon:
    package: Unknown
    path: /usr/bin/conmon
    version: 'conmon version 2.0.17, commit: 41877362fc4685d55e0473d2e4a1cbe5e1debee0'
  cpus: 8
  distribution:
    distribution: gentoo
    version: unknown
  eventLogger: file
  hostname: dellr330
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.4.38-gentoo
  linkmode: dynamic
  memFree: 2792194048
  memTotal: 8132182016
  ociRuntime:
    name: runc
    package: Unknown
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc10
      commit: dc9208a3303feef5b3839f4323d9beb36df0a9dd
      spec: 1.0.1-dev
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  rootless: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 25111392256
  swapTotal: 25769787392
  uptime: 797h 38m 40.69s (Approximately 33.21 days)
registries:
  search:
  - docker.io
  - quay.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 2
    paused: 0
    running: 1
    stopped: 1
  graphDriverName: overlay
  graphOptions:
    overlay.ignore_chown_errors: "false"
  graphRoot: /space/podman/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 836
  runRoot: /space/podman/run
  volumePath: /space/podman/volumes
version:
  APIVersion: 1
  Built: 1594213560
  BuiltTime: Wed Jul  8 13:06:00 2020
  GitCommit: 201c9505b88f451ca877d29a73ed0f1836bb96c7
  GoVersion: go1.14.2
  OsArch: linux/amd64
  Version: 2.0.2
```

**Comment by srcshelton** [comment]
Something related to https://github.com/containers/podman/commit/c4023a9302c81f04865646d765caf58ccf556cae?

**Comment by mheon** [comment]
Confirmed. Only happens on the v2.0 branch, not master.

**Comment by mheon** [comment]
Fix in #6910

**Comment by zhangguanzhang** [comment]
@rhatdan please close this issue

**Comment by TomSweeneyRedHat** [comment]
Closing as the PR addressing this has merged:  #6910
